go/pkg/reader/reader.go

```diff
@@ -209,33 +209,34 @@ func (cfs *compressedSeeker) Read(p []byte) (int, error) {
 	// to plan" error. EOF implies our uncompressed input data has finished reading,
 	// but it doesn't mean that we've finished preparing/returning our compressed data.
 	// We only want to return EOF to the user when the _compressed_ data is finished reading.
+	var finalErr error
 	if errR != nil && errR != io.EOF {
-		cfs.err = errR
+		finalErr = errR
 	} else if errW != nil {
-		cfs.err = errW
+		finalErr = errW
 	}
 
 	// When the buffer ends, or in case of _any_ error, we compress all the bytes we
 	// had available. The encoder requires a Close call to finish writing compressed
 	// data smaller than zstd's window size.
 	var errC error
-	if cfs.err != nil || errR == io.EOF {
+	if finalErr != nil || errR == io.EOF {
 		errC = cfs.encdW.Encoder.Close()
 		encoders.Put(cfs.encdW)
 		cfs.encdW = nil
 	}
 
 	m, errR2 := cfs.buf.Read(p)
 
-	if cfs.err == nil {
+	if finalErr == nil {
 		if errC != nil {
```

Comment:
If you remove `errC` above, I think all the conditionals here can go away and you can write:
```
m, errR := cfs.buf.Read(p) // no need to name it errR2 since we're done with errR by now
if finalErr == nil {
  finalErr = errR
}
return m, finalErr
```

Reply:
Done.